src/realm/column_binary.hpp

```diff
@@ -200,7 +200,7 @@ inline size_t BinaryColumn::size() const noexcept
         return leaf->size();
     }
     // Non-leaf root
-    return m_array->get_bptree_size();
+    return static_cast<BpTreeNode*>(m_array.get())->get_bptree_size();
```

Comment:
I hope in some later refactoring round to be able to remove these casts alltogether